src/main/java/me/botsko/prism/serializers/items/ItemStackSerializer.java

```diff
@@ -0,0 +1,295 @@
+        }
+        // Set basics
+        data.durability = (short) ItemUtils.getItemDamage(item);
+        data.amt = item.getAmount();
+        data.material = item.getType().name();
+        final ItemMeta meta = item.hasItemMeta() ? item.getItemMeta() : null;
+        if (meta != null) {
+            data.name = meta.getDisplayName();
+        }
+        if (meta instanceof LeatherArmorMeta) {
+            final LeatherArmorMeta lam = (LeatherArmorMeta) meta;
+            data.color = lam.getColor().asRGB();
+        } else if (meta instanceof SkullMeta) {
+            final SkullMeta skull = (SkullMeta) meta;
+            if (skull.hasOwner()) {
+                data.owner = Objects.requireNonNull(skull.getOwningPlayer()).getUniqueId().toString();
+            }
+        }
+
+        // Written books
+        if (meta instanceof BookMeta) {
+            final BookMeta bookMeta = (BookMeta) meta;
+            data.by = bookMeta.getAuthor();
+            data.title = bookMeta.getTitle();
+            data.content = bookMeta.getPages().toArray(new String[0]);
+        }
+
+        // Lore
+        if (meta != null && meta.hasLore()) {
+            data.lore = Objects.requireNonNull(meta.getLore()).toArray(new String[0]);
+        }
+
+        // Enchantments
+        if (!item.getEnchantments().isEmpty()) {
+            final String[] enchs = new String[item.getEnchantments().size()];
+            int i = 0;
+            for (final Map.Entry<Enchantment, Integer> ench : item.getEnchantments().entrySet()) {
+                // This is silly
+                enchs[i] = ench.getKey().getKey().getKey() + ":" + ench.getValue();
+                i++;
+            }
+            data.enchs = enchs;
+        } else if (meta instanceof EnchantmentStorageMeta) {
+            final EnchantmentStorageMeta bookEnchantments = (EnchantmentStorageMeta) meta;
+            if (bookEnchantments.hasStoredEnchants()) {
+                if (bookEnchantments.getStoredEnchants().size() > 0) {
+                    final String[] enchs = new String[bookEnchantments.getStoredEnchants().size()];
+                    int i = 0;
+                    for (final Map.Entry<Enchantment, Integer> ench : bookEnchantments.getStoredEnchants().entrySet()) {
+                        // This is absolutely silly
+                        enchs[i] = ench.getKey().getKey().getKey() + ":" + ench.getValue();
+                        i++;
+                    }
+                    data.enchs = enchs;
+                }
+            }
+        }
+        if (meta instanceof FireworkEffectMeta) {
+            applyFireWorkMetaToActionData(meta, data);
+        }
+        if (meta instanceof BannerMeta) {
+            List<Pattern> patterns = ((BannerMeta) meta).getPatterns();
+            Map<String, String> stringyPatterns = new HashMap<>();
+            patterns.forEach(
+                    pattern -> stringyPatterns.put(pattern.getPattern().getIdentifier(), pattern.getColor().name()));
```

Comment:
Codacy found an issue: ['lambda arguments' has incorrect indentation level 20, expected level should be one of the following: 16, 18.](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=6368169)